dispute.go

```diff
@@ -160,33 +160,37 @@ type File struct {
 // UnmarshalJSON handles deserialization of a Dispute.
 // This custom unmarshaling is needed because the resulting
 // property may be an id or the full struct if it was expanded.
-func (t *Dispute) UnmarshalJSON(data []byte) error {
+func (d *Dispute) UnmarshalJSON(data []byte) error {
+	if id, ok := ParseID(data); ok {
+		d.ID = id
+		return nil
+	}
+
 	type dispute Dispute
-	var dd dispute
-	err := json.Unmarshal(data, &dd)
-	if err == nil {
-		*t = Dispute(dd)
-	} else {
-		// the id is surrounded by "\" characters, so strip them
-		t.ID = string(data[1 : len(data)-1])
+	var v dispute
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
 	}
 
+	*d = Dispute(v)
 	return nil
 }
 
 // UnmarshalJSON handles deserialization of a File.
 // This custom unmarshaling is needed because the resulting
 // property may be an id or the full struct if it was expanded.
 func (f *File) UnmarshalJSON(data []byte) error {
```

Comment:
ooc why doesn't this live in `fileupload.go` instead?
Or maybe why isn't Dispute using FileUpload instead of File?

Reply:
Haha, I don't know! That's where I found it.

I think probably what happened is that originally files were only needed on disputes, so they were just implemented in the same file. When `FileUpload` was added, there was just no attempt to reconcile them.

For now I've moved `File` and associated tests into `fileupload.go` to make it a little easier to find in bb5e87a. I don't want to reconcile the two PRs here because `MIMEType` changing to `Type` is technically a breaking change, but that should be easy enough to do separately (too bad we didn't catch it before V33 this morning!).

Sound good?